### DISCUSSION TITLE
qual a melhor forma de instalar o mysql sem bugs?

### DISCUSSION THREAD

**Discussion by igorpinhh** [frontendbr/forum#1520] [Perguntas e Respostas]
depois de formatar o pc algumas vezes por causa de bugs sem solução e mal funcionamento do mysql eu queria saber a opinião de vocês se existe alguma forma segura/simples de se usar mysql a fim de manter o bom funcionamento dele...

**Comment by yuritoledo** [comment]
Qual SO você usa?
Talvez a melhor forma seja usando uma docker que já venha com ele instalado.

**Comment by igorpinhh** [comment]
to usando windows 10 pro, eu n conheço docker afundo mas precisa ter o mysql instalado no pc e no docker?

**Comment by yuritoledo** [comment]
Você não precisa ter o mysql instalado na sua máquina, apenas o docker mesmo.
Ele vai servir (grosseiramente) como uma máquina virtual. Você vai acessar o mysql nela, assim não precisa instalar.

Windows é meio ruim para desenvolvimento mesmo hehe.

Dá uma olhada na documentação oficial do docker, lá deve ter o que precisa.

Segue alguns links:

https://dev.mysql.com/doc/mysql-installation-excerpt/5.5/en/docker-mysql-getting-started.html
https://share.atelie.software/subindo-um-banco-de-dados-mysql-e-phpmyadmin-com-docker-642be41f7638
https://imasters.com.br/data/utilizando-docker-com-mysql

**Comment by willycamargo** [comment]
Recomendo dar uma estudada sobre docker. Acho qeu coisa de 1h é suficiente pra entender o básico caso já tenha um conhecimento prévio de algumas coisas.

**Comment by caiquealves224** [comment]
Cara, só não estranhe em transferir bugs e frustrações do MySQL para o docker, docker no windows é bem volátil, mas na pior das hipóteses, só reiniciar o docker.

**Comment by MarcosEllys** [comment]
Olá @igorpinhh , cara existe algumas formas de instalar o mysql, se for pra desenvolvimento e você for programar com PHP e Mysql vai de [XAMP](https://www.apachefriends.org/pt_br/download.html) pois eu ainda considero a opção mais fácil.

**Comment by eliseumds** [comment]
Usa Docker, mas não esquece de definir um volume pra não perder todos os dados caso o container vá pro saco.

```sh
docker run --name meu-mysql-server -v mysql-data:/var/lib/mysql -e MYSQL_ROOT_PASSWORD=minha-senha -d mysql:8
```

Isso vai gerar um diretório chamado `mysql-data` onde quer que você rode o comando, e é lá que o MySQL vai persistir os dados. Ali no final você pode escolher a versão, que no meu caso foi `8`, mas poderia ser `5`, `5.7`, o que você  precisar. Há mais instruções em https://hub.docker.com/_/mysql
